Tasks/Common/azure-arm-rest/azure-arm-resource.ts

```diff
@@ -245,19 +246,24 @@ export class Deployments {
                     resolve(new azureServiceClient.ApiResult(azureServiceClient.ToError(response)));
                 }
                 else {
-                    this.client.getLongRunningOperationResult(response).then((operationResponse) => {
-                        this.get(resourceGroupName, deploymentName, (error, response) => {
-                            if (error) {
-                                resolve(new azureServiceClient.ApiResult(error));
-                            } else {
-                                if (response.properties.provisioningState === "Succeeded") {
-                                    resolve(new azureServiceClient.ApiResult(null, response));
-                                } else {
-                                    resolve(new azureServiceClient.ApiResult(response.properties.error));
+                    this.client.getLongRunningOperationResult(response)
+                        .then((operationResponse) => {
+                            this.get(resourceGroupName, deploymentName, (error, response) => {
+                                if (error) {
+                                    resolve(new azureServiceClient.ApiResult(error));
+                                }
+                                else {
+                                    if (!response.properties) {
```

Comment:
what will be the user visible message here. I think you should just fail with a fixed message saying something to the effect "Response is not in a valid format"

Reply:
Okay